cni/pkg/nodeagent/informers.go

```diff
@@ -206,15 +206,12 @@ func (s *InformerHandlers) reconcilePod(input any) error {
 			return fmt.Errorf("failed to find namespace %v", ns)
 		}
 		wasAnnotated := oldPod.Annotations != nil && oldPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
-		isAnnotated := newPod.Annotations != nil && newPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
-		isOpOut := newPod.Annotations != nil && newPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionDisabled
+		isAnnotated := newPod.Annotations != nil && newPod.Annotations[constants.AmbientRedirection] != ""
```

Comment:
If we removed the "disabled" check, each time a updated event for the Pods(with enforced disabled) will trigger a 'remove' action, thus will introduce unexpected logs in current impl.

Reply:
That's what your PR added a test for - it passes, so with this the # of calls is the same.

It should be logically equivalent.

Reply:
My fault... I should add a more [strict test case](https://github.com/istio/istio/pull/49238/files). If we only check `isAnnotated == has ANY annotation`, whatever a Pod status update for an ambient-disabled Pod will trigger a "changeNeeded" action, that's not expected.